derive/src/max_encoded_len.rs

```
@@ -0,0 +1,203 @@
+
+	let data_expr = data_length_expr(&input.data);
+
+	quote::quote!(
+		const _: () = {
+			impl #impl_generics #mel_trait for #name #ty_generics #where_clause {
+				fn max_encoded_len() -> usize {
+					#data_expr
+				}
+			}
+		};
+	)
+	.into()
+}
+
+fn max_encoded_len_trait(input: &DeriveInput) -> syn::Result<TraitBound> {
+	let mel = {
+		const EXPECT_LIST: &str = "expect: #[max_encoded_len_mod(path::to::crate)]";
+		const EXPECT_PATH: &str = "expect: path::to::crate";
+
+		macro_rules! return_err {
+			($wrong_style:expr, $err:expr) => {
+				return Err(Error::new($wrong_style.span(), $err))
+			};
+		}
+
+		let mut mel_crates = Vec::with_capacity(2);
+		mel_crates.extend(input
+			.attrs
+			.iter()
+			.filter(|attr| attr.path == parse_quote!(max_encoded_len_mod))
+			.take(2)
+			.map(|attr| {
+				let meta_list = match attr.parse_meta()? {
+					Meta::List(meta_list) => meta_list,
+					Meta::Path(wrong_style) => return_err!(wrong_style, EXPECT_LIST),
+					Meta::NameValue(wrong_style) => return_err!(wrong_style, EXPECT_LIST),
+				};
+				if meta_list.nested.len() != 1 {
+					return_err!(meta_list, "expected exactly 1 item");
+				}
+				let first_nested =
+					meta_list.nested.into_iter().next().expect("length checked above");
+				let meta = match first_nested {
+					syn::NestedMeta::Lit(l) => {
+						return_err!(l, "expected a path item, not a literal")
+					}
+					syn::NestedMeta::Meta(meta) => meta,
+				};
+				let path = match meta {
+					Meta::Path(path) => path,
+					Meta::List(ref wrong_style) => return_err!(wrong_style, EXPECT_PATH),
+					Meta::NameValue(ref wrong_style) => return_err!(wrong_style, EXPECT_PATH),
+				};
+				Ok(path)
+			})
+			.collect::<Result<Vec<_>, _>>()?);
+
+		// we have to return `Result<Ident, Error>` here in order to satisfy the trait
+		// bounds for `.or_else` for `generate_crate_access_2018`, even though `Option<Ident>`
+		// would be more natural in this circumstance.
```

You are swallowing all errors below. So this makes no sense. Return an `Option<Result<>>` here to handle the errors.